### PR TITLE
Ability to split tests

### PR DESCRIPTION
**Pull request by prateek**

### CODE REVIEW

File: variables.sh [hunk 1]
```diff
@@ -64,3 +64,41 @@ END
...
+# pick_subset
+# $1: " " seperated string
+# $2: result var
+# $3: subset num
+# $4: subset total
+# e.g.
+#  $ pick_subset "a b c d" result 1 3
+#  $ echo $result
+#  a d
+function pick_subset()
```
Comment (richardartoul) [inline variables.sh]: pick_subset takes a space separated string and breaks it up into multiple subsets. The first argument is the space separated string, the second argument is the name of the variable to store the output in, the third variable is the subset index and the fourth variable is the desired size of each subset.

```
# pick_subset "a b c d" result 0 3
# echo $result
#  c

# pick_subset "a b c d" result 1 3
# echo $result
#  a d

# pick_subset "a b c d" result 2 3
# echo $result
#  b

Note that the generated subsets do not necessarily get grouped in the same order as the original list (as seen in the example above)

File: variables.sh [hunk 2]
```diff
@@ -64,3 +64,41 @@ END
 }
 
 export -f generate_dummy_coverage_file
+
+# pick_subset
+# $1: " " seperated string
+# $2: result var
+# $3: subset num
```
Comment (richardartoul) [inline variables.sh]: I think here (and throughout) index is a better name, then its more clear you're indexing into the list of subsets

Comment (richardartoul) [inline variables.sh]: subset_size or subset_length or something like that (also update var names)

File: variables.sh [hunk 3]
```diff
@@ -64,3 +64,41 @@ END
...
+     [[ $split_total =~ ^[0-9]+$ ]]   &&
+     (( split_num < split_total )) ; then
+       split_output=$(echo $input_to_split       \
+         | tr ' ' '\n'                           \
+         | sort                                  \
+         | awk "NR%${split_total}==${split_num}" \
+         | tr '\n' ' '                           \
+       )
+  else
+      echo "warning: illegal subset options: "  >&2
```
[the remark this answers is not in the document]
Comment (prateek) [reply variables.sh]: sure done

File: variables.sh (hunk 2 above)
Comment (prateek) [reply variables.sh]: sure

Comment (prateek) [reply variables.sh]: done

File: variables.sh (hunk 1 above)
Comment (prateek) [reply variables.sh]: done

**Pull request merged**